src/utility/index.js

```diff
@@ -0,0 +1,24 @@
+import { toast } from 'react-toastify';
+import { apiTimeOutLimit, APP_MESSAGES } from './appConstants';
+
+// eslint-disable-next-line import/prefer-default-export
+export const fetchWithTimeout = (resource, timeout = apiTimeOutLimit, options) => {
+  const controller = new AbortController();
+  setTimeout(() => controller.abort(), timeout);
+  return fetch(resource, {
+    signal: controller.signal,
+    ...options,
+  }).catch(error => {
+    toast.error(`${APP_MESSAGES.requestTimedOut}, ${error}`);
```

Comment:
Did you consider how this works with i18n context?

Reply:
The error is dynamic cannot be maintained in this context, if you have any suggestions, feel free to share that.

Reply:
Maybe we could simply add error messages as part of language files that we have, instead of different file. Because all error messages are strings inside our app.

Reply:
I have removed the service layer so this conversation is outdated, but I would like to tell you that there aren't only static messages in an application, there are browser-specific errors too, are browsers so powerful to throw an internationalized error dynamically?

Reply:
> are browsers so powerful to throw an internationalized error dynamically?

No, browsers are definitely not that powerful, yet. 🙂

I am referring to this message. 
```js
// https://github.com/pesto-students/little-tags-FE-nishantbisen09/blob/remove-es-lint-rule/src/utility/appConstants.js
export const APP_MESSAGES = {
  requestTimedOut: 'Request timed out',
};
```
This message is in our control to `i18n` it.